scripts/globals/chocobo_digging.lua

```diff
@@ -0,0 +1,255 @@
+    
+    -- 0 means we wanna wipe (probably only gonna happen onGameDay or something)
+    if (increment == 0) then
+        SetServerVariable('[DIG]ZONE'..zone..'ITEMS', 0);
+    else
+        SetServerVariable('[DIG]ZONE'..zone..'ITEMS', ZoneDigCount + increment);
+    end
+end;
+
+
+function chocoboDig(player, itemMap, precheck, messageArray, zoneWeather)
+    
+    -- make sure the player can dig before going any further
+    -- (and also cause i need a return before core can go any further with this)
+    if (precheck) then
+        return canDig(player);
+    else
+
+        local DigCount = player:getVar('[DIG]DigCount');
+        local Chance = math.random(0, 100);
+
+        if (Chance <= 15) then
+            player:messageText(player, messageArray[2]);
+            calculateSkillUp(player);
+        else
+            -- recalculate chance to compare with item abundance
+            Chance = math.random(0, 100);
+            
+            -- select a random item
+            local RandomItem = itemMap[math.random(1, #itemMap)];
+            local RItemAbundance = RandomItem[2];
+            
+            local ItemID = 0;
+            local weather = player:getWeather();
+            local moon = VanadielMoonPhase();
+            local day = VanadielDayElement();
+            
+            -- item and DIG_ABUNDANCE_BONUS 3 digits, dont wanna get left out
+            Chance = Chance * 100;
+            
+            -- We need to check for moon phase, too. 45-60% results in a much lower dig chance than the rest of the phases
+            
+            if (moon >= 45 and moon <=60) then
+              Chance = Chance * .5;
+            end
+
+            if (Chance < (RItemAbundance + DIG_ABUNDANCE_BONUS)) then
+            
+                local RItemID = RandomItem[1];
+                local RItemReq = RandomItem[3];
+                
+                -- rank 1 is burrow, rank 2 is bore (see DIGABILITY at the top of the file)
+                local DigAbility = player:getSkillRank(SKILL_DIG);
+                
+                local Mod = player:getMod(MOD_EGGHELM);
+                
+                if (RItemReq == DIGREQ_NONE) then
+                    ItemID = RItemID;
+                elseif (RItemReq == DIGREQ_BURROW and DigAbility == DIGABILITY_BURROW) then
+                    ItemID = RItemID;
+                elseif (RItemReq == DIGREQ_BORE and DigAbility == DIGABILITY_BORE) then
+                    ItemID = RItemID;
+                elseif (RItemReq == DIGREQ_MODIFIER and Mod) then
+                    ItemID = RItemID;
+                elseif (RItemReq == DIGREQ_NIGHT and VanadielTOTD() == TIME_NIGHT) then
+                    ItemID = RItemID;
+                end
+                
+                -- Let's see if the item should be obtained in this zone with this weather
+
+                if ((weather == 4 or weather == 5) and ItemID == 4096) then
```

Comment:
Honestly, nit picking over little stuff like this, that has little to do
with performance is why nobody will develop. Sure, it may not be the way
'you' do it, but as long as it's not broken, works as intended, and doesn't
cause performance issues, what's the problem with it not following your
specifications? Being so strict on little things like this will just keep
turning people away from working on anything, and will inevitably prevent
progression or result in you having to do it yourself just to meet your
expectations.

On Wednesday, March 25, 2015, starlightknight <notifications@github.com>
wrote:

> In scripts/globals/chocobo_digging.lua
> <https://github.com/DarkstarProject/darkstar/pull/1114#discussion_r27186285>
> :
>
> > +
> > +                if (RItemReq == DIGREQ_NONE) then
> > +                    ItemID = RItemID;
> > +                elseif (RItemReq == DIGREQ_BURROW and DigAbility == DIGABILITY_BURROW) then
> > +                    ItemID = RItemID;
> > +                elseif (RItemReq == DIGREQ_BORE and DigAbility == DIGABILITY_BORE) then
> > +                    ItemID = RItemID;
> > +                elseif (RItemReq == DIGREQ_MODIFIER and Mod) then
> > +                    ItemID = RItemID;
> > +                elseif (RItemReq == DIGREQ_NIGHT and VanadielTOTD() == TIME_NIGHT) then
> > +                    ItemID = RItemID;
> > +                end
> > +
> > +                -- Let's see if the item should be obtained in this zone with this weather
> > +
> > +                if ((weather == 4 or weather == 5) and ItemID == 4096) then
>
> I'm all for code readability but I don't think readability for people who
> don't understand code should be an excuse for writing bad or inefficient
> code.
>
> People who want to code on the project should be willing to learn the
> basic constructs of the programming langauge. Learning to read a table or
> an array is a pretty basic concept, and there are other strategies that can
> be used like extra line breaks for readiblity in more complex conditional
> groups to make it easier to read rather than repeating the same action over
> an over again in if/elses
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DarkstarProject/darkstar/pull/1114/files#r27186285>.
>